### ISSUE TITLE
not working

### ISSUE THREAD

**Issue by Kmfernan5** [daugaard47/ChatGPT_Conversations_To_Markdown#1]
not working

C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python -m venv venv

C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>venv\Scripts\activate

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\export chatGPT\conversations.jso
'C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\export' is not recognized as an internal or external command,
operable program or batch file.

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python json_to_markdown.py
C:\Python311\python.exe: can't open file 'C:\\Users\\djtlm\\Documents\\GitHub\\ChatGPT_Conversations_To_Markdown\\json_to_markdown.py': [Errno 2] No such file or directory

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python json_to_markdown.py
C:\Python311\python.exe: can't open file 'C:\\Users\\djtlm\\Documents\\GitHub\\ChatGPT_Conversations_To_Markdown\\json_to_markdown.py': [Errno 2] No such file or directory

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python json_to_markdown.py
Traceback (most recent call last):
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\json_to_markdown.py", line 6, in <module>
    from tqdm import tqdm
ModuleNotFoundError: No module named 'tqdm'

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>pip3 install tqdm
Collecting tqdm
  Using cached tqdm-4.65.0-py3-none-any.whl (77 kB)
Collecting colorama
  Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Installing collected packages: colorama, tqdm
Successfully installed colorama-0.4.6 tqdm-4.65.0

[notice] A new release of pip available: 22.3 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python json_to_markdown.py
Traceback (most recent call last):
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\json_to_markdown.py", line 62, in <module>
    main()
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\json_to_markdown.py", line 42, in main
    config = read_json_file(config_path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\json_to_markdown.py", line 10, in read_json_file
    data = json.load(file)
           ^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\json\__init__.py", line 293, in load
    return loads(fp.read(),
           ^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\json\decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
               ^^^^^^^^^^^^^^^^^^^^^^
json.decoder.JSONDecodeError: Invalid \escape: line 5 column 20 (char 117)

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>

**Comment by daugaard47** [comment]
@Kmfernan5 In the `config.json` file you have to give it the path to your folders/directories that holds the exported ChatGPT `.json` file. Then give it an output directory to save your markdown files.

Create the Folders/Directories and add the path here:
```
  "input_path": "C:\\YOURPATH\\ChatGPT_Conversations_To_Markdown\\JsonFiles",
  "output_directory": "C:\\YOURPATH\\ChatGPT_Conversations_To_Markdown\\MarkdownFiles",
```
You can also change this to your actual name. example:
```
"user_name": "Kmfernan5",
``

**Comment by Kmfernan5** [comment]
I did that

https://share.getcloudapp.com/Z4unOPJ1

**Comment by daugaard47** [comment]
Try Changing this:
```
  "input_path": "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\JsonFiles",
  "output_directory": "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\MarkdownFiles",
```
To this:
```
  "input_path": "C:\\Users\\djtlm\\Documents\\GitHub\\ChatGPT_Conversations_To_Markdown\\JsonFiles",
  "output_directory": "C:\\Users\\djtlm\\Documents\\GitHub\\ChatGPT_Conversations_To_Markdown\\MarkdownFiles",
```

That should solve your issue.

**Comment by Kmfernan5** [comment]
C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>venv\Scripts\activate

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python json_to_markdown.py
C:\Python311\python.exe: can't open file 'C:\\Users\\djtlm\\Documents\\GitHub\\ChatGPT_Conversations_To_Markdown\\json_to_markdown.py': [Errno 2] No such file or directory

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>C:\\Users\\djtlm\\Documents\\GitHub\\ChatGPT_Conversations_To_Markdown\\JsonFiles
'C:\\Users\\djtlm\\Documents\\GitHub\\ChatGPT_Conversations_To_Markdown\\JsonFiles' is not recognized as an internal or external command,
operable program or batch file.

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python json_to_markdown.py
Processing conversations:   0%|▏                                                     | 2/611 [00:00<00:00, 1330.47it/s]
Traceback (most recent call last):
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\json_to_markdown.py", line 62, in <module>
    main()
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\json_to_markdown.py", line 54, in main
    process_conversations(data, output_dir, config)
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\json_to_markdown.py", line 28, in process_conversations
    with open(file_path, "w", encoding="utf-8") as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 22] Invalid argument: 'C:\\Users\\djtlm\\Documents\\GitHub\\ChatGPT_Conversations_To_Markdown\\MarkdownFiles\\"Inspecting_ICE_Detention_Facilities".md'

**Comment by Kmfernan5** [comment]
How to fix?

**Comment by daugaard47** [comment]
You're on windows correct. Please tell me what your trying rather dumping the entire error

**Comment by daugaard47** [comment]
I made small update to the script and updated the README directions. This should all work fine now.

**Comment by Kmfernan5** [comment]
In my dump it shows what Im typing

**Comment by Kmfernan5** [comment]
It worked youre a genius!

**Comment by Kmfernan5** [comment]
I really wanted to use it on my betterchatGPT export but it looks like it doesn't work.

https://github.com/ztjhz/BetterChatGPT


(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python chatgpt_json_to_markdown.py
Traceback (most recent call last):
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\chatgpt_json_to_markdown.py", line 63, in <module>
    main()
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\chatgpt_json_to_markdown.py", line 54, in main
    data = read_json_file(json_file)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\chatgpt_json_to_markdown.py", line 10, in read_json_file
    data = json.load(file)
           ^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\json\__init__.py", line 293, in load
    return loads(fp.read(),
                 ^^^^^^^^^
  File "C:\Python311\Lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 201899: character maps to <undefined>

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>https://github.com/daugaard47/ChatGPT_Conversations_To_Markdown
'https:' is not recognized as an internal or external command,
operable program or batch file.

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>

**Comment by daugaard47** [comment]
Hmm.. If I find some time I'll try and create a pull request. Best of luck on your project. Looks pretty cool.
You could probably just ask ChatGPT how to add it in to your project. =)

**Comment by turnerll** [comment]
I tried that. DO you know why I get this error? 


C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python -m venv venv

C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>venv\Scripts\activate

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>pip install tqdm
Requirement already satisfied: tqdm in c:\users\djtlm\documents\github\chatgpt_conversations_to_markdown\venv\lib\site-packages (4.65.0)
Requirement already satisfied: colorama in c:\users\djtlm\documents\github\chatgpt_conversations_to_markdown\venv\lib\site-packages (from tqdm) (0.4.6)

[notice] A new release of pip available: 22.3 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip

(venv) C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown>python chatgpt_json_to_markdown.py
Processing conversations:   0%|                                                                 | 0/76 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\chatgpt_json_to_markdown.py", line 63, in <module>
    main()
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\chatgpt_json_to_markdown.py", line 55, in main
    process_conversations(data, output_dir, config)
  File "C:\Users\djtlm\Documents\GitHub\ChatGPT_Conversations_To_Markdown\chatgpt_json_to_markdown.py", line 36, in process_conversations
    content = message["content"]["parts"][0]
              ~~~~~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'parts'

**Comment by daugaard47** [comment]
Based on the error `KeyError: 'parts'` looks like `["parts"]` is empty in `message["content"]["parts"][0]` 

I don't know how to fix that for you.
